### ISSUE TITLE
Unable to parse error from a failure to execute a script

### ISSUE THREAD

**Issue by AlexITC** [sksamuel/elastic4s#1414]
elastic4s is unable to parse an error from elasticsearch.

## StackTrace
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.lang.String` out of START_ARRAY token
 at [Source: (byte[])"{"root_cause":[{"type":"remote_transport_exception","reason":"[lzIFtHT][127.0.0.1:9300][indices:data/write/update[s]]"}],"type":"illegal_argument_exception","reason":"failed to execute script","caused_by":{"type":"script_exception","reason":"runtime error","script_stack":["for (obj in params.new_values) {\n  ","                  ^---- HERE"],"script":"if (!ctx._source.containsKey(\"values\")) {\n  ctx._source.values = []\n}\nfor (obj in params.new_values) {\n  ctx._source.values.add(obj)\n}","la"[truncated 116 bytes]; line: 1, column: 273] (through reference chain: com.sksamuel.elastic4s.http.ElasticError["caused_by"]->com.sksamuel.elastic4s.http.ElasticError$CausedBy["script_stack"])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1342)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1138)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromArray(StdDeserializer.java:674)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:40)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:10)
	at com.fasterxml.jackson.databind.deser.SettableAnyProperty.deserialize(SettableAnyProperty.java:154)
	at com.fasterxml.jackson.databind.deser.SettableAnyProperty.deserializeAndSet(SettableAnyProperty.java:134)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1539)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:257)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:440)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1265)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:325)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
	at com.fasterxml.jackson.module.scala.deser.OptionDeserializer.deserialize(OptionDeserializerModule.scala:60)
	at com.fasterxml.jackson.module.scala.deser.OptionDeserializer.deserialize(OptionDeserializerModule.scala:11)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:519)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:527)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:416)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1265)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:325)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3109)
	at com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper$class.readValue(ScalaObjectMapper.scala:202)
	at com.sksamuel.elastic4s.json.JacksonSupport$$anon$1.readValue(JacksonSupport.scala:11)
	at com.sksamuel.elastic4s.http.ElasticError$.parse(ElasticError.scala:37)
	at com.sksamuel.elastic4s.http.update.UpdateImplicits$UpdateHttpExecutable$$anon$1.handle(UpdateImplicits.scala:61)
	at com.sksamuel.elastic4s.http.HttpClient$$anonfun$execute$1.apply(HttpClient.scala:69)
	at com.sksamuel.elastic4s.http.HttpClient$$anonfun$execute$1.apply(HttpClient.scala:68)
	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
	at scala.util.Try$.apply(Try.scala:192)
	at scala.util.Success.map(Try.scala:237)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

```

## How to reproduce

1. Start elasticsearch 6.2.3 on port 9200 (http)
2. Run the following code

```scala
import com.sksamuel.elastic4s.ElasticsearchClientUri
import com.sksamuel.elastic4s.http.HttpClient

import scala.util.{Failure, Success}

object ScriptTest extends App {

  import com.sksamuel.elastic4s.http.ElasticDsl._

  import scala.concurrent.ExecutionContext.Implicits.global

  val client = HttpClient.apply(ElasticsearchClientUri("localhost", 9200))

  val document = """{ "x": "y" }"""
  val id = "doc-id"

  val x = for {
    r <- indexDocument(id, document)
    _ = println(s"ELS_RESULT_1: $r")

    r <- updateWithScript(id, document)
    _ = println(s"ELS_RESULT_2: $r")
  } yield ()

  x.onComplete {
    case Success(_) =>
      client.close()

    case Failure(ex) =>
      ex.printStackTrace()
      client.close()
  }

  def indexDocument(id: String, json: String) = {
    client.execute {
      indexInto("sampleindex" / "sampletype")
          .id(id)
          .doc(json)
    }
  }

  private val UpdateScript =
    """
      |if (!ctx._source.containsKey("values")) {
      |  ctx._source.values = []
      |}
      |for (obj in params.new_values) {
      |  ctx._source.values.add(obj)
      |}
    """.stripMargin.trim

  def updateWithScript(id: String, jsonPatch: String) = {
    client.execute {
      update(id)
          .in("sampleindex" / "sampletype")
          .script(
            script(UpdateScript)
                .lang("painless")
                .param("new_values", jsonPatch)
          )
    }
  }
}
```

**Comment by t83714** [comment]
probably same issue with #1421

**Comment by sksamuel** [comment]
Fixed in 6.3.0
